apps/back/spec/App/Entity/ExpenseSpec.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace spec\App\Entity;
+
+use App\Entity\Expense;
+use App\Entity\Group;
+use App\Entity\User;
+use App\Exception\ExpenseWithoutParticipantsException;
+use DateTimeImmutable;
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+
+class ExpenseSpec extends ObjectBehavior
+{
+    function let(User $payer, User $participant, Group $group)
+    {
+        $this->beConstructedWith(
+            'dépense',
+            new DateTimeImmutable(),
+            3000,
+            new ArrayCollection([$participant->getWrappedObject(), $payer->getWrappedObject()]),
+            $payer,
+            $group
+        );
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(Expense::class);
+    }
+
+    function it_should_throw_exception_if_participant_empty(User $participant, User $payer, Group $group)
+    {
+        $this->shouldThrow(ExpenseWithoutParticipantsException::class)->during(
+            '__construct',
+            [ 'dépense',
+            new DateTimeImmutable(),
+            3000,
+            new ArrayCollection([]),
+            $payer,
+            $group]
+        );
```

Comment:
Une autre façon de faire ça c'est de déclarer ton instantiation comme d'habitude puis d'utiliser `shouldThrow(...)->duringInstanciation();`
```suggestion
        $this->beConstructedWith(
            'dépense',
            new DateTimeImmutable(),
            3000,
            new ArrayCollection([]),
            $payer,
            $group
        );
        
        $this->shouldThrow(ExpenseWithoutParticipantsException::class)->duringInstanciation();
```